vault/router.go

```diff
@@ -14,8 +14,11 @@ import (
 
 // Router is used to do prefix based routing of a request to a logical backend
 type Router struct {
-	l    sync.RWMutex
-	root *radix.Tree
+	l                    sync.RWMutex
+	root                 *radix.Tree
+	cubbyholeEntry       *routeEntry
```

Comment:
Weird to have this go through the router, bloats the scope of responsibilities. Just pass the cubbyhole directly to any callers that need it.